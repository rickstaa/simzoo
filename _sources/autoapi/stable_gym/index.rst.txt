stable_gym
==========

.. py:module:: stable_gym

.. autoapi-nested-parse::

   Entry_point that register the Stable Gym gymnasium environments.



Subpackages
-----------

.. toctree::
   :maxdepth: 1

   /autoapi/stable_gym/common/index
   /autoapi/stable_gym/envs/index


Submodules
----------

.. toctree::
   :maxdepth: 1

   /autoapi/stable_gym/version/index


Attributes
----------

.. autoapisummary::

   stable_gym.__version__
   stable_gym.__version_tuple__
   stable_gym.ENVS


Classes
-------

.. autoapisummary::

   stable_gym.MaxEpisodeStepsInjectionWrapper


Package Contents
----------------

.. py:class:: MaxEpisodeStepsInjectionWrapper(env)

   Bases: :py:obj:`gymnasium.Wrapper`


   A gymnasium wrapper that injects the ``max_episode_steps`` attribute of the
   ``TimeLimit`` wrapper into the base environment as the
   ``_time_limit_max_episode_steps`` attribute. If the environment is not wrapped in
   a ``TimeLimit`` wrapper, then the ``_time_limit_max_episode_steps`` attribute is
   set to ``None``.

   Wrap a gymnasium environment.
   :param env: The gymnasium environment.
   :type env: gym.Env


   .. py:attribute:: max_episode_steps


.. py:data:: __version__
   :value: '0.20.7'


.. py:data:: __version_tuple__

.. py:data:: ENVS

