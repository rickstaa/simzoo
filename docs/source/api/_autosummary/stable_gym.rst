stable\_gym
===========

.. automodule:: stable_gym
  
   
   
   

   
   
   

   
   
   

   
   
   



.. rubric:: Modules

.. autosummary::
   :toctree:
   :template: custom-module-template.rst
   :recursive:

   stable_gym.common
   stable_gym.envs

