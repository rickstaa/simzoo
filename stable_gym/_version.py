# Stores the package version number so that it can be accessed from other modules.
version = "0.6.4"
version_tuple = version.split(".")
