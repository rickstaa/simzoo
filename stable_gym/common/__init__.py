"""Contains several functions that can be used across all the Stable Gym gymnasium
environments.
"""
from stable_gym.common.disturber import Disturber  # noqa: F401
