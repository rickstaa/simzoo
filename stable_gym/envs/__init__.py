"""The Stable Gym gymnasium environments."""
