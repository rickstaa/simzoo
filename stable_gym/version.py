# Stores the package version number so that it can be accessed from other modules.
__version__ = "0.6.38"
__version_tuple__ = __version__.split(".")
