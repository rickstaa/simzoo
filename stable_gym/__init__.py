"""Module that register the Stable Gym gymnasium environments."""
import gymnasium as gym
from gymnasium.envs.registration import register

# Make module version available.
from .version import __version__, __version_tuple__

# Available environments.
# TODO: Update reward thresholds.
ENVS = {
    "Oscillator-v1": {
        "module": "stable_gym.envs.biological.oscillator.oscillator:Oscillator",
        "max_step": 400,
        "reward_threshold": 300,
    },
    "OscillatorComplicated-v1": {
        "module": "stable_gym.envs.biological.oscillator_complicated.oscillator_complicated:OscillatorComplicated",
        "max_step": 400,
        "reward_threshold": 300,
    },
    "CartPoleCost-v1": {
        "module": "stable_gym.envs.classic_control.cartpole_cost.cartpole_cost:CartPoleCost",
        "max_step": 250,
        "reward_threshold": 300,
    },
    "Ex3EKF-v1": {
        "module": "stable_gym.envs.classic_control.ex3_ekf.ex3_ekf:Ex3EKF",
        "max_step": 400,
        "reward_threshold": 300,
    },
    "AntCost-v1": {
        "module": "stable_gym.envs.mujoco.ant_cost.ant_cost:AntCost",
        "max_step": 250,
        "reward_threshold": 300,
    },
    "HalfCheetahCost-v1": {
        "module": "stable_gym.envs.mujoco.half_cheetah_cost.half_cheetah_cost:HalfCheetahCost",
        "max_step": 200,
        "reward_threshold": 300,
    },
    "HopperCost-v1": {
        "module": "stable_gym.envs.mujoco.hopper_cost.hopper_cost:HopperCost",
        "max_step": 250,
        "reward_threshold": 300,
    },
    "SwimmerCost-v1": {
        "module": "stable_gym.envs.mujoco.swimmer_cost.swimmer_cost:SwimmerCost",
        "max_step": 250,
        "reward_threshold": 300,
    },
    "Walker2dCost-v1": {
        "module": "stable_gym.envs.mujoco.walker2d_cost.walker2d_cost:Walker2dCost",
        "max_step": 250,
        "reward_threshold": 300,
    },
}

for env, val in ENVS.items():
    if env not in gym.envs.registry:  # NOTE: Required because we use namespace packages
        register(
            id=env,
            entry_point=val["module"],
            max_episode_steps=val["max_step"],
            reward_threshold=val["reward_threshold"],
        )
