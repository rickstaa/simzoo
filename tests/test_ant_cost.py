"""Test if the AntCost environment still behaves like the original Ant
environment when the same environment parameters are used.
"""
import gymnasium as gym
import numpy as np
from gymnasium.logger import ERROR

import stable_gym  # noqa: F401

gym.logger.set_level(ERROR)


class TestAntCostEqual:
    # Make original Ant environment.
    env = gym.make("Ant")
    # Make AntCost environment.
    env_cost = gym.make("AntCost")

    def test_equal_reset(self):
        """Test if reset behaves the same."""
        # Perform reset and check if observations are equal.
        observation, _ = self.env.reset(seed=42)
        observation_cost, _ = self.env_cost.reset(seed=42)
        assert np.allclose(
            observation, observation_cost
        ), f"{observation} != {observation_cost}"

    def test_equal_steps(self):
        """Test if steps behave the same."""
        # Perform several steps and check if observations are equal.
        self.env.reset(seed=42), self.env_cost.reset(seed=42)
        for _ in range(10):
            self.env.action_space.seed(42)
            action = self.env.action_space.sample()
            observation, _, _, _, _ = self.env.step(action)
            observation_cost, _, _, _, _ = self.env_cost.step(action)
            assert np.allclose(
                observation, observation_cost
            ), f"{observation} != {observation_cost}"
