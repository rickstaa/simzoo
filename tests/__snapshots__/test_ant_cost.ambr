# serializer version: 1
# name: TestAntCostEqual.test_snapshot
  0.8217195839822765
# ---
# name: TestAntCostEqual.test_snapshot.1
  0.9916088870480807
# ---
# name: TestAntCostEqual.test_snapshot.10
  0.02877302401613291
# ---
# name: TestAntCostEqual.test_snapshot.100
  -1.24556150010968
# ---
# name: TestAntCostEqual.test_snapshot.101
  0.16984153088564236
# ---
# name: TestAntCostEqual.test_snapshot.102
  1.0606087742013692
# ---
# name: TestAntCostEqual.test_snapshot.103
  -0.07783995342046181
# ---
# name: TestAntCostEqual.test_snapshot.104
  -0.007412282726553351
# ---
# name: TestAntCostEqual.test_snapshot.105
  -1.4813785484872442
# ---
# name: TestAntCostEqual.test_snapshot.106
  -0.19146117827004797
# ---
# name: TestAntCostEqual.test_snapshot.107
  0.44189712582356255
# ---
# name: TestAntCostEqual.test_snapshot.108
  -0.16582214242917448
# ---
# name: TestAntCostEqual.test_snapshot.109
  0.0920255602309421
# ---
# name: TestAntCostEqual.test_snapshot.11
  0.064552322654166
# ---
# name: TestAntCostEqual.test_snapshot.110
  -2.457212293528775
# ---
# name: TestAntCostEqual.test_snapshot.111
  0.3740353790349211
# ---
# name: TestAntCostEqual.test_snapshot.112
  -2.6253982609919047
# ---
# name: TestAntCostEqual.test_snapshot.113
  1.495059282386685
# ---
# name: TestAntCostEqual.test_snapshot.114
  -4.127453640396519
# ---
# name: TestAntCostEqual.test_snapshot.115
  -0.6998141706281641
# ---
# name: TestAntCostEqual.test_snapshot.116
  1.1459908578542728
# ---
# name: TestAntCostEqual.test_snapshot.117
  1.2840714156430724
# ---
# name: TestAntCostEqual.test_snapshot.118
  False
# ---
# name: TestAntCostEqual.test_snapshot.119
  False
# ---
# name: TestAntCostEqual.test_snapshot.12
  -0.011317160234533774
# ---
# name: TestAntCostEqual.test_snapshot.120
  dict({
    'cost_contact': 0.0,
    'cost_ctrl': 0.00031271414756774905,
    'cost_velocity': 1.2840714156430724,
    'distance_from_origin': 0.039746261732944946,
    'forward_reward': -0.13316874985285068,
    'penalty_health': 0.0,
    'reward_ctrl': -0.00031271414756774905,
    'reward_forward': -0.13316874985285068,
    'reward_survive': 1.0,
    'x_position': 0.03803898269853474,
    'x_velocity': -0.13316874985285068,
    'y_position': -0.011523936697341676,
    'y_velocity': 0.01790082217759864,
  })
# ---
# name: TestAntCostEqual.test_snapshot.121
  0.7516132074521321
# ---
# name: TestAntCostEqual.test_snapshot.122
  0.9955152230737171
# ---
# name: TestAntCostEqual.test_snapshot.123
  -0.07250829915548242
# ---
# name: TestAntCostEqual.test_snapshot.124
  0.05776417497617656
# ---
# name: TestAntCostEqual.test_snapshot.125
  0.018849065530907715
# ---
# name: TestAntCostEqual.test_snapshot.126
  0.5287432731470202
# ---
# name: TestAntCostEqual.test_snapshot.127
  1.0434176471790408
# ---
# name: TestAntCostEqual.test_snapshot.128
  -0.548686034880311
# ---
# name: TestAntCostEqual.test_snapshot.129
  -0.6629857014510564
# ---
# name: TestAntCostEqual.test_snapshot.13
  -0.08592924628832382
# ---
# name: TestAntCostEqual.test_snapshot.130
  0.5506670809476554
# ---
# name: TestAntCostEqual.test_snapshot.131
  -1.187663480279202
# ---
# name: TestAntCostEqual.test_snapshot.132
  0.23868478094969311
# ---
# name: TestAntCostEqual.test_snapshot.133
  1.0054166144382228
# ---
# name: TestAntCostEqual.test_snapshot.134
  -0.6785716055077514
# ---
# name: TestAntCostEqual.test_snapshot.135
  -1.3679979703014575
# ---
# name: TestAntCostEqual.test_snapshot.136
  -0.4262577253470868
# ---
# name: TestAntCostEqual.test_snapshot.137
  0.5922646014408929
# ---
# name: TestAntCostEqual.test_snapshot.138
  -0.060830609728925025
# ---
# name: TestAntCostEqual.test_snapshot.139
  -0.46177783652795623
# ---
# name: TestAntCostEqual.test_snapshot.14
  0.036875078408249884
# ---
# name: TestAntCostEqual.test_snapshot.140
  -1.1591599788878681
# ---
# name: TestAntCostEqual.test_snapshot.141
  -8.360973148991114
# ---
# name: TestAntCostEqual.test_snapshot.142
  -0.7825743516323485
# ---
# name: TestAntCostEqual.test_snapshot.143
  0.14675051316034304
# ---
# name: TestAntCostEqual.test_snapshot.144
  -0.8775617312033328
# ---
# name: TestAntCostEqual.test_snapshot.145
  4.733934123185799
# ---
# name: TestAntCostEqual.test_snapshot.146
  3.4059118403095274
# ---
# name: TestAntCostEqual.test_snapshot.147
  -3.3303330857523914
# ---
# name: TestAntCostEqual.test_snapshot.148
  2.400532615782733
# ---
# name: TestAntCostEqual.test_snapshot.149
  False
# ---
# name: TestAntCostEqual.test_snapshot.15
  -0.0958882600828999
# ---
# name: TestAntCostEqual.test_snapshot.150
  False
# ---
# name: TestAntCostEqual.test_snapshot.151
  dict({
    'cost_contact': 0.0,
    'cost_ctrl': 0.00032458479404449465,
    'cost_velocity': 2.400532615782733,
    'distance_from_origin': 0.06746068343702584,
    'forward_reward': -0.5493652299515223,
    'penalty_health': 0.0,
    'reward_ctrl': -0.00032458479404449465,
    'reward_forward': -0.5493652299515223,
    'reward_survive': 1.0,
    'x_position': 0.010570721200958626,
    'x_velocity': -0.5493652299515223,
    'y_position': -0.06662734921248344,
    'y_velocity': -1.1020682503028352,
  })
# ---
# name: TestAntCostEqual.test_snapshot.152
  0.711804146583538
# ---
# name: TestAntCostEqual.test_snapshot.153
  0.9963505762429837
# ---
# name: TestAntCostEqual.test_snapshot.154
  -0.060154520415819604
# ---
# name: TestAntCostEqual.test_snapshot.155
  0.05862134643858404
# ---
# name: TestAntCostEqual.test_snapshot.156
  0.015182247381218667
# ---
# name: TestAntCostEqual.test_snapshot.157
  0.461192022926435
# ---
# name: TestAntCostEqual.test_snapshot.158
  0.5218558172778258
# ---
# name: TestAntCostEqual.test_snapshot.159
  -0.5355688013037536
# ---
# name: TestAntCostEqual.test_snapshot.16
  0.08784503013072725
# ---
# name: TestAntCostEqual.test_snapshot.160
  -0.664580105855161
# ---
# name: TestAntCostEqual.test_snapshot.161
  0.4081626405583841
# ---
# name: TestAntCostEqual.test_snapshot.162
  -0.8945362267959178
# ---
# name: TestAntCostEqual.test_snapshot.163
  0.3408814031878626
# ---
# name: TestAntCostEqual.test_snapshot.164
  0.7955419578424723
# ---
# name: TestAntCostEqual.test_snapshot.165
  -0.44465854287553574
# ---
# name: TestAntCostEqual.test_snapshot.166
  -1.2840644698375567
# ---
# name: TestAntCostEqual.test_snapshot.167
  -1.1466432654309724
# ---
# name: TestAntCostEqual.test_snapshot.168
  0.6082945225158629
# ---
# name: TestAntCostEqual.test_snapshot.169
  0.004755707466935599
# ---
# name: TestAntCostEqual.test_snapshot.17
  -0.00499259109862529
# ---
# name: TestAntCostEqual.test_snapshot.170
  0.4510491407033329
# ---
# name: TestAntCostEqual.test_snapshot.171
  -1.5467694666176905
# ---
# name: TestAntCostEqual.test_snapshot.172
  -10.256484879948953
# ---
# name: TestAntCostEqual.test_snapshot.173
  0.33011263324733986
# ---
# name: TestAntCostEqual.test_snapshot.174
  -0.20968361887164502
# ---
# name: TestAntCostEqual.test_snapshot.175
  -4.78987517526913
# ---
# name: TestAntCostEqual.test_snapshot.176
  6.976444902220714
# ---
# name: TestAntCostEqual.test_snapshot.177
  0.6984667957870248
# ---
# name: TestAntCostEqual.test_snapshot.178
  -5.049843744069792
# ---
# name: TestAntCostEqual.test_snapshot.179
  2.4002850772702193
# ---
# name: TestAntCostEqual.test_snapshot.18
  -0.018486236354526057
# ---
# name: TestAntCostEqual.test_snapshot.180
  False
# ---
# name: TestAntCostEqual.test_snapshot.181
  False
# ---
# name: TestAntCostEqual.test_snapshot.182
  dict({
    'cost_contact': 0.0,
    'cost_ctrl': 0.0001541744112968445,
    'cost_velocity': 2.4002850772702193,
    'distance_from_origin': 0.13407571151076497,
    'forward_reward': -0.5492853440442207,
    'penalty_health': 0.0,
    'reward_ctrl': -0.0001541744112968445,
    'reward_forward': -0.5492853440442207,
    'reward_survive': 1.0,
    'x_position': -0.01689354600125241,
    'x_velocity': -0.5492853440442207,
    'y_position': -0.13300715965925083,
    'y_velocity': -1.327596208935348,
  })
# ---
# name: TestAntCostEqual.test_snapshot.19
  -0.06809295444039414
# ---
# name: TestAntCostEqual.test_snapshot.2
  -0.0774271412280302
# ---
# name: TestAntCostEqual.test_snapshot.20
  0.12225413386740303
# ---
# name: TestAntCostEqual.test_snapshot.21
  -0.015452948206880215
# ---
# name: TestAntCostEqual.test_snapshot.22
  -0.042832782216310726
# ---
# name: TestAntCostEqual.test_snapshot.23
  -0.03521335504882296
# ---
# name: TestAntCostEqual.test_snapshot.24
  0.05323091855533488
# ---
# name: TestAntCostEqual.test_snapshot.25
  0.036544406436407835
# ---
# name: TestAntCostEqual.test_snapshot.26
  0.04127326115959884
# ---
# name: TestAntCostEqual.test_snapshot.27
  dict({
  })
# ---
# name: TestAntCostEqual.test_snapshot.28
  0.8407919712982543
# ---
# name: TestAntCostEqual.test_snapshot.29
  0.9932166193724319
# ---
# name: TestAntCostEqual.test_snapshot.3
  0.09074426673746092
# ---
# name: TestAntCostEqual.test_snapshot.30
  -0.07483860895325636
# ---
# name: TestAntCostEqual.test_snapshot.31
  0.08112274236978242
# ---
# name: TestAntCostEqual.test_snapshot.32
  0.0365927627099295
# ---
# name: TestAntCostEqual.test_snapshot.33
  0.18401997564096487
# ---
# name: TestAntCostEqual.test_snapshot.34
  0.44059221183872344
# ---
# name: TestAntCostEqual.test_snapshot.35
  -0.15625536523131922
# ---
# name: TestAntCostEqual.test_snapshot.36
  -0.3810127897954648
# ---
# name: TestAntCostEqual.test_snapshot.37
  0.1785463166038994
# ---
# name: TestAntCostEqual.test_snapshot.38
  -0.3834662217311874
# ---
# name: TestAntCostEqual.test_snapshot.39
  0.16631644068768867
# ---
# name: TestAntCostEqual.test_snapshot.4
  0.04982299652572164
# ---
# name: TestAntCostEqual.test_snapshot.40
  0.4140176816545653
# ---
# name: TestAntCostEqual.test_snapshot.41
  -0.1962819434245767
# ---
# name: TestAntCostEqual.test_snapshot.42
  -0.17760139619696663
# ---
# name: TestAntCostEqual.test_snapshot.43
  0.12977166603540113
# ---
# name: TestAntCostEqual.test_snapshot.44
  0.15727142415211362
# ---
# name: TestAntCostEqual.test_snapshot.45
  -0.8041248650060641
# ---
# name: TestAntCostEqual.test_snapshot.46
  -1.0253112328541807
# ---
# name: TestAntCostEqual.test_snapshot.47
  5.103576027721423
# ---
# name: TestAntCostEqual.test_snapshot.48
  12.541117699928481
# ---
# name: TestAntCostEqual.test_snapshot.49
  -5.79477567722775
# ---
# name: TestAntCostEqual.test_snapshot.5
  0.05721286105539078
# ---
# name: TestAntCostEqual.test_snapshot.50
  -5.304115207359589
# ---
# name: TestAntCostEqual.test_snapshot.51
  3.731844297088785
# ---
# name: TestAntCostEqual.test_snapshot.52
  -7.4905278175382435
# ---
# name: TestAntCostEqual.test_snapshot.53
  3.9994275515215874
# ---
# name: TestAntCostEqual.test_snapshot.54
  9.033972437290057
# ---
# name: TestAntCostEqual.test_snapshot.55
  1.1788254175671897
# ---
# name: TestAntCostEqual.test_snapshot.56
  False
# ---
# name: TestAntCostEqual.test_snapshot.57
  False
# ---
# name: TestAntCostEqual.test_snapshot.58
  dict({
    'cost_contact': 0.0,
    'cost_ctrl': 0.00031490790843963625,
    'cost_velocity': 1.1788254175671897,
    'distance_from_origin': 0.052401502902274535,
    'forward_reward': -0.08573726912508142,
    'penalty_health': 0.0,
    'reward_ctrl': -0.00031490790843963625,
    'reward_forward': -0.08573726912508142,
    'reward_survive': 1.0,
    'x_position': 0.0505043462549386,
    'x_velocity': -0.08573726912508142,
    'y_position': -0.013972419825440287,
    'y_velocity': -0.034962155517014905,
  })
# ---
# name: TestAntCostEqual.test_snapshot.59
  0.8276472520968416
# ---
# name: TestAntCostEqual.test_snapshot.6
  -0.07437727346489083
# ---
# name: TestAntCostEqual.test_snapshot.60
  0.9952936880306066
# ---
# name: TestAntCostEqual.test_snapshot.61
  -0.07252653421509932
# ---
# name: TestAntCostEqual.test_snapshot.62
  0.05708549138589211
# ---
# name: TestAntCostEqual.test_snapshot.63
  0.029523263275106286
# ---
# name: TestAntCostEqual.test_snapshot.64
  0.3864590962855272
# ---
# name: TestAntCostEqual.test_snapshot.65
  1.208452412969746
# ---
# name: TestAntCostEqual.test_snapshot.66
  -0.3879966443933651
# ---
# name: TestAntCostEqual.test_snapshot.67
  -0.5219557890182384
# ---
# name: TestAntCostEqual.test_snapshot.68
  0.339316363732256
# ---
# name: TestAntCostEqual.test_snapshot.69
  -0.8484633293460893
# ---
# name: TestAntCostEqual.test_snapshot.7
  -0.009922812420886579
# ---
# name: TestAntCostEqual.test_snapshot.70
  0.22569464625828592
# ---
# name: TestAntCostEqual.test_snapshot.71
  0.8366010116440885
# ---
# name: TestAntCostEqual.test_snapshot.72
  -0.04709091270090322
# ---
# name: TestAntCostEqual.test_snapshot.73
  0.20592219087103156
# ---
# name: TestAntCostEqual.test_snapshot.74
  -0.7042417608831626
# ---
# name: TestAntCostEqual.test_snapshot.75
  -0.06495336401841977
# ---
# name: TestAntCostEqual.test_snapshot.76
  -0.8870544293366875
# ---
# name: TestAntCostEqual.test_snapshot.77
  0.3718878625380356
# ---
# name: TestAntCostEqual.test_snapshot.78
  3.0121673925802104
# ---
# name: TestAntCostEqual.test_snapshot.79
  18.12550445638559
# ---
# name: TestAntCostEqual.test_snapshot.8
  -0.025840395153483756
# ---
# name: TestAntCostEqual.test_snapshot.80
  -3.4942168466123245
# ---
# name: TestAntCostEqual.test_snapshot.81
  -0.539023411367595
# ---
# name: TestAntCostEqual.test_snapshot.82
  2.709549200894454
# ---
# name: TestAntCostEqual.test_snapshot.83
  -11.081676080943627
# ---
# name: TestAntCostEqual.test_snapshot.84
  -1.583948806863844
# ---
# name: TestAntCostEqual.test_snapshot.85
  7.876496394579283
# ---
# name: TestAntCostEqual.test_snapshot.86
  1.245765198674449
# ---
# name: TestAntCostEqual.test_snapshot.87
  False
# ---
# name: TestAntCostEqual.test_snapshot.88
  False
# ---
# name: TestAntCostEqual.test_snapshot.89
  dict({
    'cost_contact': 0.0,
    'cost_ctrl': 0.0002168221950531006,
    'cost_velocity': 1.245765198674449,
    'distance_from_origin': 0.04639062816451278,
    'forward_reward': -0.11613852127522639,
    'penalty_health': 0.0,
    'reward_ctrl': -0.0002168221950531006,
    'reward_forward': -0.11613852127522639,
    'reward_survive': 1.0,
    'x_position': 0.04469742019117728,
    'x_velocity': -0.11613852127522639,
    'y_position': -0.012418977806221608,
    'y_velocity': 0.031068840384373582,
  })
# ---
# name: TestAntCostEqual.test_snapshot.9
  0.08535299776972036
# ---
# name: TestAntCostEqual.test_snapshot.90
  0.7714324261248123
# ---
# name: TestAntCostEqual.test_snapshot.91
  0.9942783641856469
# ---
# name: TestAntCostEqual.test_snapshot.92
  -0.08593684662780433
# ---
# name: TestAntCostEqual.test_snapshot.93
  0.060763057860797214
# ---
# name: TestAntCostEqual.test_snapshot.94
  0.018254963801364432
# ---
# name: TestAntCostEqual.test_snapshot.95
  0.5721321075810565
# ---
# name: TestAntCostEqual.test_snapshot.96
  1.305099251586962
# ---
# name: TestAntCostEqual.test_snapshot.97
  -0.46466445692869085
# ---
# name: TestAntCostEqual.test_snapshot.98
  -0.601339741376164
# ---
# name: TestAntCostEqual.test_snapshot.99
  0.5929510739274422
# ---
